### PR TITLE
Gudhi citation update

### LINKED ISSUE #101: Old dates in `how_to_cite_gudhi.bib`
@book{gudhi:urm
, title        = "{GUDHI} User and Reference Manual"
, author      = "{The GUDHI Project}"
, publisher     = "{GUDHI Editorial Board}"
, year         = 2015
, url =    "http://gudhi.gforge.inria.fr/doc/latest/"
}

Shouldn't this have `year=2019`? I would actually update all the dates in the file at every release.

### LINKED ISSUE #102: Citing a specific gudhi version
`how_to_cite_gudhi.bib` shows how to link to the "current" (for an evolving definition of current) version of gudhi and of each package (assuming we never go renaming packages since that would break links).
On the other hand, CGAL, in their `how_to_cite_cgal.bib`, tells you how to cite a specific release. For people writing a paper, I believe it is important that they specify which version of gudhi they are using. I think I prefer the CGAL way, to be discussed.
This is not completely independent of #101.

### PR DESCRIPTION
**Pull request by VincentRouvreau**
. Fix #101 
. Fix #102

### CODE REVIEW

File: biblio/how_to_cite_gudhi.bib [hunk 1]
```diff
@@ -2,141 +2,157 @@ @book{gudhi:urm
...
 
 @incollection{gudhi:FilteredComplexes
 , author =  "Cl\'ement Maria"
 , title =   "Filtered Complexes"
 , publisher =  "{GUDHI Editorial Board}"
+, edition =     "{3.0.0}"
 , booktitle =   "{GUDHI} User and Reference Manual"
-, url = "http://gudhi.gforge.inria.fr/doc/latest/group__simplex__tree.html"
-, year =        2015
+, url = "https://gudhi.inria.fr/doc/3.0.0/group__simplex__tree.html"
```
Comment (mglisse) [inline biblio/how_to_cite_gudhi.bib]: Hmm, now we will have to remember to update this before each release (the next release isn't going to be 3.0.0)... I guess we could generate this file from cmake so the release is automatically inserted. Do you have other ideas?

Comment (VincentRouvreau) [reply biblio/how_to_cite_gudhi.bib]: And what about the CGAL one ? Do we have to update it at each release or not ?
Your idea of bibtex generation is a really good idea. I will have a look at it.

Comment (mglisse) [reply biblio/how_to_cite_gudhi.bib]: > And what about the CGAL one ? Do we have to update it at each release or not ?

Yeah, I don't know about that one, seems less important and I can't think of a very good solution. Maybe just update it once in a while, when we remember?

File: biblio/how_to_cite_gudhi.bib.in [hunk 2]
```diff
@@ -2,141 +2,157 @@ @book{gudhi:urm
 , title        = "{GUDHI} User and Reference Manual"
 , author      = "{The GUDHI Project}"
 , publisher     = "{GUDHI Editorial Board}"
-, year         = 2015
-, url =    "http://gudhi.gforge.inria.fr/doc/latest/"
+, edition =     "{@GUDHI_VERSION@}"
+, year         = 2019
```
Comment (mglisse) [inline biblio/how_to_cite_gudhi.bib.in]: Can't we also generate the year, while we are there?

File: biblio/how_to_cite_gudhi.bib (hunk 1 above)
Comment (VincentRouvreau) [reply biblio/how_to_cite_gudhi.bib]: It is now generated for GUDHI on e1404be9da9cda8e6f41e7cf207934f1914a3505

File: biblio/how_to_cite_gudhi.bib.in (hunk 2 above)
Comment (VincentRouvreau) [reply biblio/how_to_cite_gudhi.bib.in]: Year is now also generated 518de123bcf19a85accca5660844355da1106511

**Pull request merged**